### PR TITLE
martian/header: optimized ViaModifier implementation

### PR DESCRIPTION
**Pull request by mmatczuk**

### CODE REVIEW

File: internal/martian/header/via_modifier.go [hunk 1]
```diff
@@ -49,42 +55,36 @@ func NewViaModifier(requestedBy string) *ViaModifier {
...
+		l += len(via) + 2
+	}
+
+	var sb strings.Builder
+	sb.Grow(l)
 
-	if v := req.Header.Get("Via"); v != "" {
-		if m.hasLoop(v) {
+	if via != "" {
+		if strings.Contains(via, m.requestedBy+"-"+m.boundary) {
```
Comment (Choraden) [inline internal/martian/header/via_modifier.go:69]: How about precomputing `m.requestedBy+"-"+m.boundary` instead of concatenating on every modify?

**Comment by Choraden** [comment]
LGTM.

I feel that optimisations are correct, but it would be useful to have #676.

File: internal/martian/header/via_modifier.go (hunk 1 above)
Comment (mmatczuk) [reply internal/martian/header/via_modifier.go:69]: Sure.

**Comment by mmatczuk** [comment]
@Choraden check the new implementation

**Comment by mmatczuk** [comment]
Here we could have a microbenchmark

**Comment by Choraden** [comment]
> Here we could have a microbenchmark

That's true.

**Comment by mmatczuk** [comment]
```
 michal@SL-2094:martian/header (mmt/has_loop2)$ benchstat old.txt new.txt
goos: darwin
goarch: arm64
pkg: github.com/saucelabs/forwarder/internal/martian/header
cpu: Apple M1 Pro
            │    old.txt    │               new.txt               │
            │    sec/op     │    sec/op     vs base               │
ViaModifier   2237.0n ± ∞ ¹   152.8n ± ∞ ¹  -93.17% (p=0.029 n=4)
¹ need >= 6 samples for confidence interval at level 0.95

            │   old.txt    │              new.txt               │
            │     B/op     │    B/op      vs base               │
ViaModifier   1440.0 ± ∞ ¹   160.0 ± ∞ ¹  -88.89% (p=0.029 n=4)
¹ need >= 6 samples for confidence interval at level 0.95

            │   old.txt    │              new.txt               │
            │  allocs/op   │  allocs/op   vs base               │
ViaModifier   31.000 ± ∞ ¹   4.000 ± ∞ ¹  -87.10% (p=0.029 n=4)
¹ need >= 6 samples for confidence interval at level 0.95
```

**Comment by mmatczuk** [comment]
It was indeed slow

**Review by Choraden — APPROVED** [review]
💪

**Pull request merged**